python/ql/lib/semmle/python/ApiGraphs.qll

```diff
@@ -485,6 +485,40 @@ module API {
       )
     }
 
+    /**
+     * Holds if `result` is the result of awaiting `awaitedValue`.
+     */
+    cached
+    DataFlow::Node awaited(DataFlow::Node awaitedValue) {
+      // `await` x
+      // - `awaitedValue` is `x`
+      // - `result` is `await x`
+      exists(Await await |
+        result.asExpr() = await and
+        await.getValue() = awaitedValue.asExpr()
+      )
+      or
+      // `async for x in l`
+      // - `awaitedValue` is `l`
+      // - `result` is `l` (should perhaps be `x`, but that should really be a read)
+      exists(AsyncFor asyncFor |
+        result.asExpr() = asyncFor.getTarget() and
+        // Morally, we should perhaps use asyncFor.getIter() = awaitedValue.asExpr(),
+        // but that is actually behind a read step rather than a flow step.
+        asyncFor.getTarget() = awaitedValue.asExpr()
+      )
```

Comment:
without having a test it's hard to really understand whether this modeling is correct... I don't really know if a `for` would make much sense in a normal (non-async) api graph usage. :thinking: (since it goes to the elements of the iterable `l`, which I don't think we model in any API graph related way right now).